azurerm/internal/services/purview/purview_account_resource.go

```diff
@@ -0,0 +1,302 @@
+package purview
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/purview/mgmt/2020-12-01-preview/purview"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/location"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/purview/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/purview/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourcePurviewAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePurviewAccountCreateUpdate,
+		Read:   resourcePurviewAccountRead,
+		Update: resourcePurviewAccountCreateUpdate,
+		Delete: resourcePurviewAccountDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.AccountName(),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"sku_capacity": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ValidateFunc: validation.IntInSlice([]int{
+					4,
+					16,
+				}),
+			},
+
+			"sku_name": {
```

Comment:
@allantargino Sorry for being unclear there, I’ve been chatting to @katbyte a few weeks back on how we define SKU, basically I believe we want `sku_name` to contain both name and capacity and parse them inline, i.e. `sku_name` property will allow for `Standard_4` and `Standard_16` as values.

@katbyte that was the conclusion, right?

Reply:
yes typically thats how we've been handling the sku names as of late

Reply:
Oh I see, let me refactor that then. Thanks for the inputs :)